### PR TITLE
Feature/case visits

### PR DESCRIPTION
**Pull request by gilleswittenberg**
Work in progress...

### CODE REVIEW

File: src/app/state/rest/CacheStore.ts [hunk 1]
```diff
@@ -48,8 +48,7 @@ class CacheStore {
 const store = new CacheStore()
 
 if (process.env.NODE_ENV === "development") {
-  // @ts-ignore
-  window.cacheStore = store
+  (window as { cacheStore?: CacheStore }).cacheStore = store
```
Comment (gilleswittenberg) [inline src/app/state/rest/CacheStore.ts:51]: This causes merge conflict. Can be removed.

File: src/app/features/caseVisits/components/organisms/Form/Fields.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { useForm } from "react-final-form"
+import createScaffold from "./scaffold"
+import ScaffoldFields from "app/features/shared/components/molecules/Form/ScaffoldFields"
+import type { CaseVisit } from "../../../hooks/useCaseVisits"
+
+type Props = {
+  initialValues?: CaseVisit
+}
+
+const Fields: React.FC<Props> = ({ initialValues }) => {
```
Comment (gilleswittenberg) [inline src/app/features/caseVisits/components/organisms/Form/Fields.tsx:11]: This part probably needs a review the most

File: src/app/features/cases/components/organisms/TableCases/TableCases.tsx [hunk 3]
```diff
@@ -10,24 +10,27 @@ import { useCases } from "app/state/rest/config"
...
+  { minWidth: 120 }
 ]
 
 const mapData = (data: API.Case) => [
   data.start_date ?? "-",
   data.end_date ?? "-",
   data.address.street_name ?? "-",
-  <OpenButton href={to("/cases/:id", { id: data.id })} />
+  <OpenButton href={to("/cases/:id", { id: data.id })} text="open" />,
+  <OpenButton href={to("/case-visits/:caseId", { caseId: data.id })} text="bezoeken" />
```
Comment (Jeffrey-Zutt) [inline src/app/features/cases/components/organisms/TableCases/TableCases.tsx:33]: Might be better to put both `OpenButton` components in the same column:
Only the last column is sticky on mobile.

Comment (gilleswittenberg) [reply src/app/features/cases/components/organisms/TableCases/TableCases.tsx:33]: Ok. Will push.

File: src/app/features/caseVisits/components/organisms/Form/Fields.tsx (hunk 2 above)
Comment (Jeffrey-Zutt) [reply src/app/features/caseVisits/components/organisms/Form/Fields.tsx:11]: I think its better to create a custom form-component: "AutoFillButton" or something.

```
type Props = React.ComponentProps<typeof Button> & {
  field: string
  target: string
}

const AutoFillButton:React.FC<Props> = ({ field, target, ...restProps }) => {
  const [ isDisabled, setIsDisabled ] = useState(false)
  const { getFieldState, change } = useForm()

  const handleClick = useCallback(() => {
     
    const value = [
      getFieldState(target)?.value ?? "", 
      getFieldState(field)?.value ?? ""
    ].filter(_ => _ !== "").join("\n")
    
    change("note", value)
    setIsDisabled(true)
    
  }, [ getFieldState, change, field, target ])

  return <Button {...restProps} onClick={handleClick} disabled={isDisabled}>
}
```

Register it in: 
`src/app/features/shared/components/molecules/Form/ScaffoldField.tsx`

And call it in createScaffold as:
```
note0Button: {
      type: "AutoFillButton",
      props: {
        label: "Voeg toe",
        align: "right",
        field: "note0",
        target: "note"
      }
    }
```

That way we have (unit testable) reusable component we could use elsewhere as well.

Comment (Jeffrey-Zutt) [reply src/app/features/caseVisits/components/organisms/Form/Fields.tsx:11]: Also: 
I think we should be careful passing functions as props without wrapping them in a `useCallback`

**Comment by Jeffrey-Zutt** [comment]
Left some small remarks! But looks awesome already!

File: src/app/features/caseVisits/components/organisms/Form/Fields.tsx (hunk 2 above)
Comment (gilleswittenberg) [reply src/app/features/caseVisits/components/organisms/Form/Fields.tsx:11]: Looks cleaner, yes

**Pull request merged**